cmd/snap-update-ns/bootstrap.c

```diff
@@ -247,6 +247,89 @@ int validate_snap_name(const char* snap_name)
     return 0;
 }
 
+static int instance_key_validate(const char *instance_key)
+{
+    // NOTE: see snap.ValidateInstanceName for reference of a valid instance key
+    // format
+
+    // Ensure that name is not NULL
+    if (instance_key == NULL) {
+        bootstrap_msg = "instance key cannot be NULL";
+        return -1;
+    }
+
+    const char *p = instance_key;
```

Comment:
A note here like in validate_snap_name would be nice. Eg:
```
// This is a regexp-free routine hand-codes the following pattern:
//
// "^[a-z]{1,10}$"
//
// The only motivation for not using regular expressions is so that we
// don't run untrusted input against a potentially complex regular
// expression engine.
```
```